pkg/fanal/analyzer/language/php/composer/composer.go

```diff
@@ -2,41 +2,81 @@ package composer
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
+	"io/fs"
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 
 	"golang.org/x/xerrors"
 
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
 	"github.com/aquasecurity/go-dep-parser/pkg/php/composer"
+	godeptypes "github.com/aquasecurity/go-dep-parser/pkg/types"
 	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
 	"github.com/aquasecurity/trivy/pkg/fanal/analyzer/language"
 	"github.com/aquasecurity/trivy/pkg/fanal/types"
-	"github.com/aquasecurity/trivy/pkg/fanal/utils"
+	"github.com/aquasecurity/trivy/pkg/log"
+	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
 )
 
 func init() {
-	analyzer.RegisterAnalyzer(&composerLibraryAnalyzer{})
+	analyzer.RegisterPostAnalyzer(types.Composer, newComposerAnalyzer)
 }
 
 const version = 1
 
-var requiredFiles = []string{types.ComposerLock}
+type composerAnalyzer struct {
+	lockParser godeptypes.Parser
+}
+
+func newComposerAnalyzer(_ analyzer.AnalyzerOptions) (analyzer.PostAnalyzer, error) {
+	return &composerAnalyzer{
+		lockParser: composer.NewParser(),
+	}, nil
+}
+
+func (a composerAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysisInput) (*analyzer.AnalysisResult, error) {
+	var apps []types.Application
 
-type composerLibraryAnalyzer struct{}
+	required := func(path string, d fs.DirEntry) bool {
+		return filepath.Base(path) == types.ComposerLock
+	}
+
+	err := fsutils.WalkDir(input.FS, ".", required, func(path string, d fs.DirEntry, r dio.ReadSeekerAt) error {
+		// Parse composer.lock
+		app, err := a.parseComposerLock(path, r)
+		if err != nil {
+			return xerrors.Errorf("parse error: %w", err)
+		} else if app == nil {
+			return nil
+		}
 
-func (a composerLibraryAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
-	res, err := language.Analyze(types.Composer, input.FilePath, input.Content, composer.NewParser())
+		// Parse composer.json alongside composer.lock to identify the direct dependencies
+		if err = a.mergeComposerJson(input.FS, filepath.Dir(path), app); err != nil {
+			return err
+		}
+		sort.Slice(app.Libraries, func(i, j int) bool {
+			return app.Libraries[i].ID < app.Libraries[j].ID
+		})
+		apps = append(apps, *app)
 
+		return nil
+	})
 	if err != nil {
-		return nil, xerrors.Errorf("error with composer.lock: %w", err)
+		return nil, xerrors.Errorf("composer walk error: %w", err)
 	}
-	return res, nil
+
+	return &analyzer.AnalysisResult{
+		Applications: apps,
+	}, nil
 }
 
-func (a composerLibraryAnalyzer) Required(filePath string, _ os.FileInfo) bool {
+func (a composerAnalyzer) Required(filePath string, _ os.FileInfo) bool {
 	fileName := filepath.Base(filePath)
-	if !utils.StringInSlice(fileName, requiredFiles) {
+	if fileName != types.ComposerLock && fileName != types.ComposerJson {
```

Comment:
```suggestion
	return slices.Contains(requiredFiles, fileName)
```

Reply:
We can't use `return` here, because after that we check `vendor` dir, but i used `requiredFiles` in that check.